src/extensions/replay/sessionrecording.ts

```diff
@@ -434,19 +433,31 @@ export class SessionRecording {
         this.sessionId = sessionId
     }
 
-    private _tryTakeFullSnapshot(): boolean {
-        // TODO this should ignore based on emit?
+    private _tryTakeFullSnapshot(withCustomEvent?: { tag: string; payload: any }): void {
         if (!this._captureStarted) {
-            return false
+            return
         }
-        try {
-            this.rrwebRecord?.takeFullSnapshot()
-            return true
-        } catch (e) {
-            // Sometimes a race can occur where the recorder is not fully started yet, so we can't take a full snapshot.
-            logger.error('Error taking full snapshot.', e)
-            return false
+        // try the same action three times 50 milliseconds apart
+        // if it fails all three times then we give up
+        // this is to account for the fact that the recorder might not be started yet
+        let tries = 0
+        const retryingFn = () => {
+            tries++
+            if (tries > 3) {
+                logger.warn('Could not take full snapshot after 150 milliseconds. Giving up.')
+                return
+            }
+
+            try {
+                if (withCustomEvent) {
+                    this.rrwebRecord?.addCustomEvent(withCustomEvent.tag, withCustomEvent.payload)
+                }
+                this.rrwebRecord?.takeFullSnapshot()
+            } catch (e) {
+                setTimeout(retryingFn, 50)
+            }
         }
+        retryingFn()
```

Comment:
I don't get this change... If the recorder hasn't started - that's okay right? We wouldn't take a full snapshot, as the starting of the recorder will take the snapshot.

It feels unnecessary and the real fix is in having the customEvent come _after_ the `_captureStarted` check?

Reply:
Oh, that's where I was heading, but we already had take full snapshot wrapped in a check to swallow failing to take one. So, the code was written to drop the request for a full snapshot. Which feels wrong

Reply:
and this is on return from idle so everything should already be running

but, let's swallow the error the same way and get the error out of sentry. it's only a debug signal